src/sptps.c

```diff
@@ -128,11 +128,13 @@ static bool send_record_priv(sptps_t *s, uint8_t type, const void *data, uint16_
 
 // Send an application record.
 bool sptps_send_record(sptps_t *s, uint8_t type, const void *data, uint16_t len) {
+
+	if((!s->outstate) && (type != PKT_PROBE))
+		return error(s, EINVAL, "Handshake phase not finished yet (outstate:%x type: %d)",
```

Comment:
You can of course remove the outstate, type information in this commit - tinc-ponyhof does have excesive debugging informations.